src/vm/jitinterface.cpp

```diff
@@ -7053,6 +7053,53 @@ bool getILIntrinsicImplementationForUnsafe(MethodDesc * ftn,
         methInfo->options = (CorInfoOptions)0;
         return true;
     }
+    else if (tk == MscorlibBinder::GetMethod(METHOD__UNSAFE__BYREF_READ_UNALIGNED)->GetMemberDef() ||
+             tk == MscorlibBinder::GetMethod(METHOD__UNSAFE__PTR_READ_UNALIGNED)->GetMemberDef())
+    {
+        _ASSERTE(ftn->HasMethodInstantiation());
+        Instantiation inst = ftn->GetMethodInstantiation();
+        _ASSERTE(ftn->GetNumGenericMethodArgs() == 1);
+        mdToken tokGenericArg = FindGenericMethodArgTypeSpec(MscorlibBinder::GetModule()->GetMDImport());
```

Comment:
`FindGenericMethodArgTypeSpec` finds a typespec token that points to `!!0` signature. This signature is context free. The same typespec token is shared between all methods that reference it, so it will work as long as there is at least one method that references `!!0` in CoreLib.

> but SizeOf does not

SizeOf happens to work because of there is other method that references `!!0` signature. If that other place was changed or deleted, SizeOf would break. It would be nice to add `typeof(T).ToString()` to SizeOf as well.